### ISSUE TITLE
Add is_deallocate_trivial to Allocators.Background

### ISSUE THREAD

**Issue by vinniefalco** [CPPAlliance/json#257]
it is a distinguishing feature of boost.json

**Comment by grisumbras** [comment]
I don't think that Allocators.Background is the place for this, because it describes standard allocators and PMR, rather then JSON's approach. Meanwhile, [Allocators.storage_ptr](https://www.boost.org/doc/libs/1_79_0/libs/json/doc/html/json/allocators/storage_ptr.html) does mention the trait. Do we need something more elaborate?
